52_wa_diff_shipment.py

```diff
@@ -0,0 +1,263 @@
+
+other_states_result = (
+    other_states_shipment_copy.groupby(["Year", "StateName", "county_test"])[
+        "shipment_per_100k"
+    ]
+    .sum()
+    .reset_index()
+)
+other_states_result.head()
+
+washington_summary = pd.DataFrame(
+    washington_prescriptions_result.describe()["shipment_per_100k"]
+).rename(columns={"shipment_per_100k": "Opioid Shipment per 100k Residents - Texas"})
+other_states_summary = pd.DataFrame(
+    other_states_result.describe()["shipment_per_100k"]
+).rename(
+    columns={"shipment_per_100k": "Opioid Shipment per 100k Residents - Comp States"}
+)
+stats = pd.concat([washington_summary, other_states_summary], axis=1)
+stats
+# calculating yearwise drug related deaths results for Oregon
+Oregon_results = (
+    other_states_shipment_copy[other_states_shipment_copy["StateName"] == "Oregon"]
+    .groupby("Year")["shipment_per_100k"]
+    .mean()
+)
+Oregon_results
+# calculating yearwise drug related deaths results for Idaho
+Idaho_results = (
+    other_states_shipment_copy[other_states_shipment_copy["StateName"] == "Idaho"]
+    .groupby("Year")["shipment_per_100k"]
+    .mean()
+)
+Idaho_results
+# calculating yearwise drug related deaths results for Montana
+montana_results = (
+    other_states_shipment_copy[other_states_shipment_copy["StateName"] == "Montana"]
+    .groupby("Year")["shipment_per_100k"]
+    .mean()
+)
+montana_results
+# calculating yearwise drug related deaths results for Nevada
+nevada_results = (
+    other_states_shipment_copy[other_states_shipment_copy["StateName"] == "Nevada"]
+    .groupby("Year")["shipment_per_100k"]
+    .mean()
+)
+nevada_results
+
+wa_result = (
+    washington_prescriptions_result.groupby("Year")["shipment_per_100k"]
+    .mean()
+    .reset_index()
+)
+comp_result = (
+    other_states_result.groupby(["Year"])["shipment_per_100k"].mean().reset_index()
+)
+comp_result
+wa_result
+
+florida_summary = pd.DataFrame(wa_result.describe()["shipment_per_100k"]).rename(
+    columns={"shipment_per_100k": "Opioid Shipment per 100k Residents - Washington"}
+)
+comp_summary = pd.DataFrame(comp_result.describe()["shipment_per_100k"]).rename(
+    columns={"shipment_per_100k": "Opioid Shipment per 100k Residents - Control States"}
+)
+stats = pd.concat([florida_summary, comp_summary], axis=1)
+stats
+# create a scale for number of years before and after 2012 (target year)
+
+
+def scale_years(year):
+    if year == 2009:
+        return -2
+    if year == 2010:
+        return -1
+    if year == 2011:
+        return 0
+    if year == 2012:
+        return 1
+
+
+wa_result["year relative to policy"] = wa_result["Year"].apply(lambda x: scale_years(x))
+comp_result["year relative to policy"] = comp_result["Year"].apply(
+    lambda x: scale_years(x)
+)
+
+# double check no nulls in "year relative to policy"
+
+assert wa_result["year relative to policy"].isnull().sum() == 0
+assert comp_result["year relative to policy"].isnull().sum() == 0
+
```

Comment:
This code will need to be deleted, since we are not required to scale down the year values with respect to 2012 (for eg: 2010 - (-2), 2011 - (-1) and so on). We are supposed to show the years in the plots as they are - eg: 2010, 2011 and so on.

Reply:
I think also we need to apply new dataset re-uploaded by Alex!

Reply:
Yes, we need to, Zhonglin!